lib/matplotlib/layout_engine.py

```diff
@@ -101,6 +101,28 @@ def execute(self, fig):
         raise NotImplementedError
 
 
+class NullLayoutEnigne(LayoutEngine):
+    """
+    A no-op LayoutEngine.
+
+    This is primarily to act as a place holder if we remove a layout engine.
+
+    Parameter
+    ---------
+    adjust_compatible, colorbar_gridspec : bool
+        Allow the NullLayoutEnigne to mirror the behavior of whatever
+        layout engine it is replacing.
+
+    """
+    def __init__(self, adjust_compatible, colorbar_gridspec, **kwargs):
+        self._adjust_compatible = adjust_compatible
+        self._colorbar_gridspec = colorbar_gridspec
+        super().__init__(**kwargs)
+
+    def execute(self, fig):
```

Comment:
I guess I'm not sure about adding a Null engine here.  It makes `set_layout_engine('none')` go through a (slightly) different codepath than never setting the layout engine at all.  If we do this, we should probably initialize the figure with this NullLayoutEngine?  But I'm not sure why we want to have this at all.

Reply:
We can not stick a formatter on at init time without making the bools tri-states (`{unset, True, False}`).  This acts as a no-op place holder that remember the colorbar related settings without inventing another side-band way to store that.

Unfortunately I think that "never been set" vs "was set and then removed" are in fact different.

Reply:
Ah I see - we don't want to switch colorbar behaviour after it has started to be used...  So we are turning off the algorithm, but keeping its side effect. 

At some point we need to figure out how to ditch the two ways of placing colorbars without breaking everybody.